R/src_gen.R

```diff
@@ -0,0 +1,207 @@
+#' @title Generate Synthetic Rating Curve
+#' @param comids A set of COMIDs. See \link{find_comids}.
+#' @param hand A HAND \code{raster}.
+#' @param stage A set of stage values. Defaults to \code{0:20}.
+#' @param progress If \code{TRUE}, output a progress bar.
+#' @return A \code{data.frame} representing the SRC table.
+#' @importFrom nhdplusTools get_vaa
+#' @importFrom data.table :=
+#' @importFrom plyr .
+#' @export
+get_src <- function(comids, hand, stage = 0:20, progress = TRUE) {
+    if (progress) {
+        pb <- progress::progress_bar$new(
+            total = length(stage) + 3,
+            format = " generating SRC [:bar] :percent eta: :eta"
+        )
+
+        pb$tick(0)
+    }
+
+    if (missing(comids) & !missing(hand)) {
+        hand_bb <- sf::st_bbox(hand) %>%
+                   sf::st_as_sfc() %>%
+                   sf::st_as_sf()
+
+        comids <- find_comids(hand_bb)
+    }
+
+    # If the HAND raster is not passed, get from AWS
+    if (missing(hand) & !missing(comids)) {
+        nhd <- nhdplusTools::get_nhdplus(comid = comids)
+        hand <- get_hand_raster(nhd)
+    }
+
+    # Fix NA values
+    hand[is.na(hand[])] <- 0
+    # Generate Slope Raster
+    slope <- raster::terrain(hand)
```

Comment:
We need to be careful here, we need to make sure this is slope in %, I dont think terrain offers that....

Reply:
`raster::terrain()` can be replaced with `sf::gdal_utils(util = "demprocessing", ...)` I believe. However, I think the scale is off when this function is used. The GDAL documentation states that the option `-s ...` can be used to modify the scale, but `sf::gdal_utils()` is returning an error when this option is included.

```
# Generate Slope Raster
tmp <- tempfile(fileext = ".tif")

sf::gdal_utils(util = "demprocessing",
               source = hand@file@name,
               destination = tmp,
               processing = "slope",
               options = c("-p", "-s SCALE_VALUE"))

slope <- raster::raster(tmp)
slope[is.na(slope[])] <- 0
unlink(tmp)
```